src/weather_magic/core.cljs

```diff
@@ -37,8 +37,24 @@
       (gl/make-buffers-in-spec state/gl-ctx glc/static-draw)
       (cam/apply camera)))
 
+(defn align-animation
+  "Function that handles alignment or zoom-alignment"
+  []
+  (let [delta-x (:delta-x @state/pointer-zoom-info)
+        delta-y (:delta-y @state/pointer-zoom-info)
+        delta-fov (:delta-fov @state/pointer-zoom-info)
+        total-steps (:total-steps @state/pointer-zoom-info)
+        current-step (:current-step @state/pointer-zoom-info)
+        delta-z-angle (:delta-z-angle @state/pointer-zoom-info)]
+    (event-handlers/update-zoom-point-alignment delta-x delta-y delta-z-angle current-step delta-fov)
+    (swap! state/pointer-zoom-info assoc-in [:current-step] (inc current-step))
+    (when (= current-step total-steps)
+      (swap! state/pointer-zoom-info assoc :state false))))
+
 (defn draw-frame! [t]
   (when (and @(:loaded @state/base-texture) @(:loaded (:trump @state/textures)))
+    (when (:state @state/pointer-zoom-info)
+      (align-animation))
```

Comment:
Varför stoppar vi inte bara in den här funktionen i `@state/earth-animation-fn`? Vi vill ju att den ska avbrytas om användaren börjar scrolla eller på annat vis modifiera kamerans position - inte fortsätta zooma in trots att användaren scrollar ut, eller?